Source/ParameterEncoding.swift

```diff
@@ -151,7 +151,11 @@ public enum ParameterEncoding {
     }
 
     func escape(string: String) -> String {
-        let legalURLCharactersToBeEscaped: CFStringRef = ":&=;+!@#$()',*"
+        let generalDelimiters = ":#[]@" // dropping "?" and "/" due to RFC 3986 - Section 3.4
```

Comment:
Ya makes sense. May be better to document why those aren't there, rather than saying "we're dropping them". "We're dropping them sounds more like a temp comment, rather than documentation expected to live there for a long time. Make sense?

And does that mean the square bracket tests were failing before this PR?